src/core/network/qgsnetworkaccessmanager.cpp

```diff
@@ -681,6 +737,23 @@ void QgsNetworkAccessManager::setupDefaultProxyAndCache( Qt::ConnectionType conn
 
   if ( cache() != newcache )
     setCache( newcache );
+
+  if ( sMainNAM )
+  {
+    static_cast<QgsNetworkCookieJar *>( cookieJar() )->setAllCookies( static_cast<QgsNetworkCookieJar *>( sMainNAM->cookieJar() )->allCookies() );
```

Comment:
This isn't thread safe -- it's accessing allCookies from the main thread's object in a background thread

Reply:
Should be addressed by protecting all QgsNetworkCookieJar methods with a mutex.